packages/cli/src/packages-to-be/fee-currency.ts

```diff
@@ -0,0 +1,126 @@
+import { feeCurrencyWhitelistABI } from '@celo/abis'
+import { feeCurrencyDirectoryABI } from '@celo/abis-12/FeeCurrencyDirectory'
+import { StrongAddress } from '@celo/base'
+import { getContract, PublicClient, Transport } from 'viem'
+import { celo } from 'viem/chains'
+import { ContractAddressResolver } from './address-resolver'
+
+const MINIMAL_TOKEN_INFO_ABI = [
+  {
+    type: 'function' as const,
+    stateMutability: 'view',
+    outputs: [{ type: 'string', name: '', internalType: 'string' }],
+    name: 'symbol',
+    inputs: [],
+  },
+  {
+    type: 'function' as const,
+    stateMutability: 'view',
+    outputs: [{ type: 'string', name: '', internalType: 'string' }],
+    name: 'name',
+    inputs: [],
+  },
+  {
+    type: 'function' as const,
+    stateMutability: 'view',
+    inputs: [],
+    outputs: [{ type: 'address', name: '', internalType: 'address' }],
+    name: 'adaptedToken',
+  },
+  //  usdt adapter uses slightly different interface this adaptedToken/getAdaptedToken are aliases
+  {
+    type: 'function' as const,
+    stateMutability: 'view',
+    inputs: [],
+    outputs: [{ type: 'address', name: '', internalType: 'address' }],
+    name: 'getAdaptedToken',
+  },
+  {
+    type: 'function' as const,
+    stateMutability: 'view',
+    inputs: [],
+    outputs: [{ type: 'uint8', name: '', internalType: 'uint8' }],
+    name: 'decimals',
+  },
+] as const
+
+export interface FeeCurrencyInformation {
+  name?: string
+  symbol?: string
+  address: StrongAddress
+  adaptedToken?: StrongAddress
+  decimals?: number
+}
+
+export interface FeeCurrencyProvider {
+  getAddresses(): Promise<StrongAddress[]>
+  getFeeCurrencyInformation(whitelist?: StrongAddress[]): Promise<FeeCurrencyInformation[]>
+}
+
+export class ViemFeeCurrencyProvider implements FeeCurrencyProvider {
```

Comment:
the issue i see with this pattern is that we would end up needing to duplicate the logic for each rpc library. 

That isnt terrible if we are just doing this temporarily for the web3 to viem transtion but seem pointless for a library we maintain long term